oneview/data_source_interconnects.go

```diff
@@ -0,0 +1,736 @@
+			},
+			"uid_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"unsupported_capabilities": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceInterconnectsRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	name := d.Get("name").(string)
+
+	interconnect, err := config.ovClient.GetInterconnectByName(name)
+	if err != nil || interconnect.URI.IsNil() {
+		d.SetId("")
+
+		return nil
+	}
+
+	d.SetId(name)
+
+	d.Set("base_wwn", interconnect.BaseWWN)
+	d.Set("category", interconnect.Category)
+	d.Set("description", interconnect.Description)
+	d.Set("device_reset_state", interconnect.DeviceResetState)
+	d.Set("eTag", interconnect.ETag)
+	d.Set("edge_virtual_bridging_available", interconnect.EdgeVirtualBridgingAvailable)
+	d.Set("enable_cut_through", interconnect.EnableCutThrough)
+	d.Set("enable_fast_mac_cache_failover", interconnect.EnableFastMacCacheFailover)
+	d.Set("enable_igmp_snooping", interconnect.EnableIgmpSnooping)
+	d.Set("enable_network_loop_protection", interconnect.EnableNetworkLoopProtection)
+	d.Set("enable_pause_flood_protection", interconnect.EnablePauseFloodProtection)
+	d.Set("enable_rich_tlv", interconnect.EnableRichTLV)
+	d.Set("enable_storm_control", interconnect.EnableStormControl)
+	d.Set("enable_tagged_lldp", interconnect.EnableTaggedLldp)
+	d.Set("enclosure_name", interconnect.EnclosureName)
+	d.Set("enclosure_type", interconnect.EnclosureType)
+	d.Set("enclosure_uri", interconnect.EnclosureUri.String())
+	d.Set("firmware_version", interconnect.FirmwareVersion)
+	d.Set("host_name", interconnect.HostName)
+	d.Set("igmp_idle_timeout_interval", interconnect.IgmpIdleTimeoutInterval)
+	d.Set("igmp_snooping_vlan_ids", interconnect.IgmpSnoopingVlanIds)
+	d.Set("interconnect_ip", interconnect.InterconnectIP)
+	d.Set("interconnect_mac", interconnect.InterconnectMAC)
+	d.Set("interconnect_type_uri", interconnect.InterconnectTypeUri.String())
+	d.Set("lldp_ip_address_mode", interconnect.LldpIpAddressMode)
+	d.Set("lldp_ipv4_address", interconnect.LldpIpv4Address)
+	d.Set("lldp_ipv6_address", interconnect.LldpIpv6Address)
+	d.Set("logical_interconnect_uri", interconnect.LogicalInterconnectUri.String())
+	d.Set("max_bandwidth", interconnect.MaxBandwidth)
+	d.Set("mgmt_interface", interconnect.MgmtInterface)
+	d.Set("migration_state", interconnect.MigrationState)
+	d.Set("model", interconnect.Model)
+	d.Set("name", interconnect.Name)
+	d.Set("network_loop_protection_interval", interconnect.NetworkLoopProtectionInterval)
+	d.Set("part_number", interconnect.PartNumber)
+	d.Set("port_count", interconnect.PortCount)
+	d.Set("power_state", interconnect.PowerState)
+	d.Set("product_name", interconnect.ProductName)
+	d.Set("scopes_uri", interconnect.ScopesUri)
+	d.Set("serial_number", interconnect.SerialNumber)
+	d.Set("spare_part_number", interconnect.SparePartNumber)
```

Comment:
I don't know enough about oneview to know for sure but it doesn't seem like this attribute provides valuable information to a user. Every attribute an api supports doesn't need to be specified inside of Terraform but if you believe that a user can find value with this information than feel free to keep it. There are quite a few attributes here that would just seem to clutter the statefile. This one datasource adds over 200 lines to the statefile and could make reading through what information Terraform is keeping track of extremely difficult and so I'd suggest paring this down to only the essential information that would benefit the user

Reply:
We are not sure about the level of value for each field for the user as we are not familiar with a lot of the use cases. Will it be fine for now to keep the existing fields and remove the unwanted ones when we know more about it.

Reply:
It's easier to add fields than remove them. It's a breaking change when you remove a field from the schema because you don't know what users are using them and removing it could break them. It's a very simple thing to add a requested field in but a painful task removing an unused field. I'll stress again that a simpler, smaller, less cluttered schema is preferred here

Reply:
@mbfrahry Thanks for the input. I have removed the fields I think are unnecessary. I don't think I can remove any more fields. If you think any more fields need to be removed please specify.

Reply:
@mbfrahry Can you please take a look at #70  as wel.